system/datatypes/peer_reviewed_article.php

```diff
@@ -416,14 +429,49 @@ public function editAction(HyphaRequest $request) {
 			return ['redirect', $this->constructFullPath($this->pagename)];
 		}
 
+		$formData = $request->getPostData() + $_FILES;
+
 		// create form
-		$form = $this->createEditForm($request->getPostData());
+		$form = $this->createEditForm($formData);
+
+		$featuredImageData = $form->dataFor(self::FIELD_NAME_FEATURED_IMAGE);
+		$orgImg = $this->xml->documentElement->getOrCreate(self::FIELD_NAME_FEATURED_IMAGE)->getText();
+		$hasNewImg = $featuredImageData && !$featuredImageData['error'];
+
+		if ($hasNewImg) {
+			$ext = strtolower(substr(strrchr($featuredImageData['name'], '.'), 1));
+			@$size = getimagesize($_FILES[self::FIELD_NAME_FEATURED_IMAGE]['tmp_name']);
+			if (!$size || !in_array($ext, ['jpg','jpeg','png','gif','bmp'])) {
+				$form->errors[self::FIELD_NAME_FEATURED_IMAGE] = __('invalid-image-file').ini_get('upload_max_filesize');
+			}
+		}
 
 		// process form if there are no errors
 		if (!empty($form->errors)) {
 			return $this->editViewRender($request, $form);
 		}
 
+		$newImage = null;
+		if ($hasNewImg) {
+			$maxSize = [1120, 800];
+			$needResize = $size[0] > $maxSize[0] || $size[1] > $maxSize[1];
+			$filename = uniqid() . '.' . $ext;
+			$destinations = ['org' => 'data/images/org/' . $filename, 'img' => 'data/images/' . $filename];
+			if ($needResize && !file_exists('data/images/org/')) {
+				mkdir('data/images/org/', 0777, true);
+			}
+			$orgUrlIndex = $needResize ? 'org' : 'img';
+			if (move_uploaded_file($_FILES[self::FIELD_NAME_FEATURED_IMAGE]['tmp_name'], $destinations[$orgUrlIndex])) {
+				if ($needResize) {
+					image_resize($destinations[$orgUrlIndex], $destinations['img'], $maxSize[0], $maxSize[1]);
+				}
+			}
```

Comment:
This still ignores a failure of `move_uploaded_file` or `image_resize` (though that might trigger a PHP error, of course, which might be sufficien).